### PR TITLE
close #11330 sets uses optimized countSetBits

### LINKED ISSUE #11330: sets use non-optimized countBits32
countbits32

/cc @krux02 

related #10617

### PR DESCRIPTION
**Pull request by ringabout**
close #11330

followup
- [ ] now that sets is an import module, move it to lib/std/private/sets_builtin
https://github.com/nim-lang/Nim/pull/17334#discussion_r598067587

- [ ] ditto for countbits_impl

### CODE REVIEW

File: lib/system/sets.nim
Comment (timotheecour) [inline lib/system/sets.nim]: is this comment about `code duplication` still relevant? which code duplication?

Comment (ringabout) [reply lib/system/sets.nim]: make sense

Comment (ringabout) [reply lib/system/sets.nim]: fixed

File: lib/system.nim
Comment (Araq) [inline lib/system.nim]: This is blocking the merge until my basic IC tests are merged. Yet another system module dependency...

Comment (ringabout) [reply lib/system.nim]: ok

File: lib/pure/bitops.nim
Comment (timotheecour) [inline lib/pure/bitops.nim]: * now that sets is an import module, move it to `lib/std/private/sets_builtin`

this way, we also avoid 2 import modules with same name:
lib/pure/collections/sets.nim
system/sets

which can cause issues (eg for docs https://nim-lang.github.io/Nim/sets.html, references etc)

* this comment also needs to be upgrade to reflect this and fact that it's import, not include

File: lib/system/countbits_impl.nim
Comment (timotheecour) [inline lib/system/countbits_impl.nim]: move runnableExamples to bitops.countSetBits

File: lib/system/countbits_impl.nim
Comment (timotheecour) [inline lib/system/countbits_impl.nim]:
```suggestion
```

**Review by timotheecour — APPROVED** [review]
LGTM after addressing comments

File: lib/pure/bitops.nim
Comment (konsumlamm) [reply lib/pure/bitops.nim]: For consistency with `channels_builtin` (#17330), perhaps use `sets_builtin`?

Comment (timotheecour) [reply lib/pure/bitops.nim]: sure, i updated the comment

Comment (ringabout) [reply lib/pure/bitops.nim]: `countbits_impl` can also be moved to `std/private`, so I left it the following PR.

File: lib/system/countbits_impl.nim
Comment (ringabout) [reply lib/system/countbits_impl.nim]: fixed

**Review by timotheecour — APPROVED** [review]

File: lib/pure/bitops.nim
Comment (Araq) [inline lib/pure/bitops.nim:36]:
```suggestion
```

Comment (Araq) [inline lib/pure/bitops.nim]:
```suggestion
```

File: lib/system.nim
Comment (Araq) [inline lib/system.nim]: Undo this change please, `system/sets` is now only an include file

Comment (ringabout) [reply lib/system.nim]: OK, will finish it later

File: lib/system/sets.nim
Comment (ringabout) [inline lib/system/sets.nim]:
```suggestion
proc cardSet(s: NimSet, len: int): int {.compilerproc, inline.} =
```

File: lib/system/sets.nim
Comment (ringabout) [inline lib/system/sets.nim]:
```suggestion
```

File: lib/system.nim
Comment (ringabout) [inline lib/system.nim]:
```suggestion
  import system/countbits_impl
  include "system/sets"
```

**Pull request merged**
